Sprint 14

VictorMugisha opened the pull request.

VictorMugisha committed:
fix: added logs to text the frontend socket connections

InshutiSouvede committed:
chore: show no coach for rows with deactivated coaches

VictorMugisha committed:
fix: now the join waitlist updates are changed dynamically

VictorMugisha committed:
chore: updated the reserve code for Google Apps Script

VictorMugisha committed:
chore: changed the waitlist form link

InshutiSouvede committed:
chore: show the coach name for all participants who have completed the phase

InshutiSouvede committed:
Merge branch 'dev' into fix/deactivated-coach
Merge dev into fix/deactivated-coach

InshutiSouvede committed:
chore: reflect the changes from deactivating coach instantly on overview pages

InshutiSouvede committed:
chore: resolve eslint errors

VictorMugisha committed:
Merge branch 'dev' of https://github.com/kb21-rw/trainee-app-frontend into fix/waitlist-dynamic-update

InshutiSouvede committed:
show the dropdown icon on coach column and show options on a single click

InshutiSouvede committed:
chore: move the singleclick logic to a function

InshutiSouvede committed:
chore: resolve eslint errors

VictorMugisha committed:
Merge pull request #220 from kb21-rw/fix/waitlist-dynamic-update

Dynamic Update After Joining Wait List

InshutiSouvede committed:
fix: avoid showing trainee stages on the applicants overview

InshutiSouvede committed:
Merge pull request #221 from kb21-rw/fix/deactivated-coach

(Admin) label "no coach" for both applicant and trainee overview incase  a coach is deactivated.

InshutiSouvede committed:
Merge pull request #222 from kb21-rw/fix/overview-stages

(Admin & Coach) Incorrect Last Application Stage Displayed

musayann commented on src/components/applicationForm/ApplicationFormActions.tsx:
```diff
@@ -1,150 +1,155 @@
+  userStatus,
+}: ApplicationFormActionsProps) {
+  const status =
+    role === UserRole.Prospect && userStatus === UserStatus.Applied
```
You can also simplify this by 
```ts
[UserRole.Prospect, UserStatus.Applied].includes(role)
```

musayann commented on src/components/applicationForm/ApplicationFormActions.tsx:
```diff
@@ -1,150 +1,155 @@
+
+  useEffect(() => {
+    if (socket && data?.email) {
+      socket.emit("join-room", data.email)
```
I would suggest to convert the socket interactions into a hook. Something `useRoom` or `useSockets`

musayann commented on src/components/applicationForm/ApplicationFormActions.tsx:
```diff
@@ -1,150 +1,155 @@
+    if (socket && data?.email) {
+      socket.emit("join-room", data.email)
+
+      socket.on("joinedTheWaitList", (message) => {
```
follow the same naming convention for the events. It should be in `kebab-case`. I would suggest to also create an enum for the socket events.
eg:
```ts
enum SocketEvent {
   JoinRoom = 'join-room',
   JoinedWaitlist = 'joined-waitlist',
   WaitlistError = 'waitlist-error'
}
```

musayann commented on src/components/ui/OverViewTable.tsx:
```diff
@@ -211,39 +213,51 @@ export default function OverViewTable({
+
+    // Determine participant phase
+    const status = participant?.traineeStatus
+    const isCompleted = status === "GRADUATED" || isPostselectionForTrainees
```
please use enums as much as you can. Avoid hardcoding text for checking statuses or any other of condition.

VictorMugisha committed:
refactor: update socket event by replacing string literals with enumerated socket events

VictorMugisha committed:
fix: fixed prettier issues in OverViewTable

dependabot[bot] committed:
chore(deps-dev): bump vite from 6.3.4 to 6.3.6 (#230)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.3.4 to 6.3.6.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.3.6/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.3.6/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.3.6
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

InshutiSouvede committed:
Merge pull request #229 from kb21-rw/fix/sprint-14-reviews

Sprint 14 Reviews
